### PR TITLE
Better handling of invalid modules from Addons API

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In some edge cases, modules list from Addons API can be invalid: we need to ignore invalid modules or this break the Module Catalog page and every page with "Recommended modules" action.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Add an invalid module or ask Addons team to add an invalid module in their api lulz

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8831)

### CODE REVIEW

File: src/Adapter/Module/AdminModuleDataProvider.php [hunk 1]
```diff
@@ -172,9 +172,14 @@ public function isAllowedAccess($action, $name = '')
...
 
-    public function generateAddonsUrls(array $addons, $specific_action = null)
+    public function generateAddonsUrls(AddonsCollection $addons, $specific_action = null)
     {
-        foreach ($addons as &$addon) {
+        foreach ($addons->toArray() as &$addon) {
+            if (!$addon->hasValidInstance()) {
+                $this->logger->warning(sprintf("The addon with id %s is invalid", $addon->get('id')));
+
+                return;
```
Comment (Quetzacoalt91) [inline src/Adapter/Module/AdminModuleDataProvider.php]: `return` or `continue`?

Comment (mickaelandrieu) [reply src/Adapter/Module/AdminModuleDataProvider.php]: o_O continue

**Comment by Quetzacoalt91** [comment]
Aren't we supposed to have a `AddonsCollection` class somewhere is that PR? I do not remember seeing this in the project

**Comment by mickaelandrieu** [comment]
> Aren't we supposed to have a AddonsCollection class somewhere is that PR? I do not remember seeing this in the project

:+1:

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 please review this asap, I've added the missing `AddonsCollection` class  :+1:

File: src/Adapter/Module/AdminModuleDataProvider.php [hunk 2]
```diff
@@ -172,8 +172,9 @@ public function isAllowedAccess($action, $name = '')
         return ($this->employee->can('edit', 'AdminModulessf') && $this->moduleProvider->can('configure', $name));
     }
 
-    public function generateAddonsUrls(array $addons, $specific_action = null)
+    public function generateAddonsUrls(AddonsCollection $addons, $specific_action = null)
     {
+        $addons = $addons->toArray();
```
Comment (Quetzacoalt91) [inline src/Adapter/Module/AdminModuleDataProvider.php:177]: Is this line needed when you implement `Iterator` in the concerned class? :)

Comment (mickaelandrieu) [reply src/Adapter/Module/AdminModuleDataProvider.php:177]: i prefer being explicit on what I retrieve (a simple array)

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
This PR fixes an issue on the module page on `develop` branch, but the QA may confirm there is no regression. :)

**Comment by Quetzacoalt91** [comment]
Thank you @mickaelandrieu

**Pull request merged**